tmd/Topology/analysis.py

```diff
@@ -20,6 +20,67 @@ def sort_ph(ph):
     return np.array(ph)[np.argsort([p[0] - p[1] for p in ph])].tolist()
 
 
+def closest_ph(ph_list, target_extent, method='from_above', return_index=False):
+    """
+    Returns the barcode list that has the maximum extent
+    that is closer to the target_extent according to the selected method.
+
+    method:
+        from_above: smallest maximum extent that is greater or equal than target_extent
+        from_below: biggest maximum extent that is smaller or equal than target_extent
+        nearest: closest by absolute value
+    """
+    n_bars = len(ph_list)
+
+    max_extents = \
+    np.fromiter((max(get_lengths(barcode)) for barcode in ph_list), dtype=np.float)
```

Comment:
Any reason for not doing 
`np.array([max(get_lengths(barcode)) for barcode in ph_list], dtype=np.float)`
np.fromiter is a bit crap in term of perf I believe

Reply:
Hmm, so generating a list and converting it to a numpy array is slower than mapping a generator directly to a numpy array without extra copies? Python never seizes to amaze me :D. Anyways, this is not a matter of performance, so I can make it a tad more readable.